Formula/haproxy.rb

```diff
@@ -24,6 +27,13 @@ def install
       USE_ZLIB=1
       ADDLIB=-lcrypto
     ]
+    if build.with? "lua@5.3"
+      args << "USE_LUA=1"
+      args << "LUA_LIB_NAME=lua.5.3"
+      args << "LUA_LIB=#{Formula["lua@5.3"].lib}"
```

Comment:
Unfortunately lua@5.3 is little bit non-standard (include path and lib version suffix), only property i was able to omit is `LUA_LIB`.

`LUA_LD_FLAGS` needs to be replaced because `--export-dynamic` flag does not seem to be supported on OS X.

https://github.com/haproxy/haproxy/blob/25f067ccec52f53b0248a05caceb7841a3cb99df/Makefile#L605